core/src/test/scala/org/apache/spark/sql/delta/DeltaOptionSuite.scala

```diff
@@ -130,4 +132,16 @@ class DeltaOptionSuite extends QueryTest
           "change the partition schema").getMessage)
     }
   }
+
+  test("support the maxRecordsPerFile write option") {
+    val tempDir = Utils.createTempDir()
```

Comment:
nit: we usually use `withTempDir`. Could you write more test cases to cover different write paths, such as
```
import org.apache.spark.sql.delta.test.DeltaSQLCommandTest

class DeltaOptionSuite extends ...
  with ...
  with DeltaSQLCommandTest {
  ...

  test("support the maxRecordsPerFile write option: path") {
    withTempDir { tempDir =>
      val path = tempDir.getCanonicalPath
      withTable("maxRecordsPerFile") {
        spark.range(100)
          .write
          .format("delta")
          .option("maxRecordsPerFile", 5)
          .save(path)
        assert(FileUtils.listFiles(tempDir, Array("parquet"), false).size === 20)
      }
    }
  }

  test("support the maxRecordsPerFile write option: external table") {
    withTempDir { tempDir =>
      val path = tempDir.getCanonicalPath
      withTable("maxRecordsPerFile") {
        spark.range(100)
          .write
          .format("delta")
          .option("maxRecordsPerFile", 5)
          .option("path", path)
          .saveAsTable("maxRecordsPerFile")
        assert(FileUtils.listFiles(tempDir, Array("parquet"), false).size === 20)
      }
    }
  }

  test("support the maxRecordsPerFile write option: v2 write") {
    withTempDir { tempDir =>
      val path = tempDir.getCanonicalPath
      withTable("maxRecordsPerFile") {
        spark.range(100)
          .writeTo(s"maxRecordsPerFile")
          .using("delta")
          .option("maxRecordsPerFile", 5)
          .tableProperty("location", path)
          .create()
        assert(FileUtils.listFiles(tempDir, Array("parquet"), false).size === 20)
      }
    }
  }
}
```

Reply:
@zsxwing I appreciate the feedback. I have made the suggested change here: https://github.com/delta-io/delta/pull/1017/commits/c9d092f3c170d6793c2f33f78e9442608880fdc8 (+https://github.com/delta-io/delta/pull/1017/commits/d548e40b7137c18ef43288ee20e986afd45d7eeb)

Please provide further guidance if there are any other tests to add. I have minimal experience in this space, and do not know what other variations should be tested.